lib/ruby-mpd/parser.rb

```diff
@@ -105,25 +114,26 @@ def parse_line(line)
     # The end result is a hash containing the proper key/value pairs
     def build_hash(string)
       return {} if string.nil?
+      array_keys = {}
 
       string.lines.each_with_object({}) do |line, hash|
         key, object = parse_line(line)
 
         # if val appears more than once, make an array of vals.
         if hash.include? key
-          hash[key] = Array(hash[key]) << object
+          # cannot use Array(hash[key]) or [*hash[key]] because Time instances get splatted
+          # cannot check for is_a?(Array) because some values (time) are already arrays
+          unless array_keys[key]
```

Comment:
Can you try using `Array.wrap`? I think that one should safely work with Time objects as well, since it uses `to_ary` instead of `to_a`. http://apidock.com/rails/Array/wrap/class